src/index.js

```diff
@@ -72,5 +72,6 @@ export const denormalize = (input, schema, entities) => {
     return input;
   }
 
+  entities = { ...entities, __cache: {} };
```

Comment:
I don't like adding a reserved key here. There's no reason someone might have a `__cache` entity. (I know, it's not the best name for an entity, but I'd hate to ruin someone's day if they tried to use it).

I'd rather see a separate `visitedEntities` cache object passed around than using reserved keys.